bin/helpers/capabilityHelper.js

```diff
@@ -185,6 +196,27 @@ const validate = (bsConfig, args) => {
     } catch(error){
       reject(Constants.validationMessages.INVALID_CYPRESS_JSON)
     }
+
+    //check if home_directory is present or not in user run_settings
+    if (!Utils.isUndefined(bsConfig.run_settings.home_directory)) {
+      // check if home_directory exists or not
+      if (!fs.existsSync(bsConfig.run_settings.home_directory)) {
+        reject(Constants.validationMessages.HOME_DIRECTORY_NOT_FOUND);
+      }
+
+      // check if home_directory is a directory or not
+      if (!fs.statSync(bsConfig.run_settings.home_directory).isDirectory()) {
+        reject(Constants.validationMessages.HOME_DIRECTORY_NOT_A_DIRECTORY);
+      }
+
+      // check if cypress config file (cypress.json) is a part of home_directory or not
+      if (!path.resolve(bsConfig.run_settings.cypressConfigFilePath).includes(path.resolve(bsConfig.run_settings.home_directory))) {
```

Comment:
have we tested whether the special charaters in path names or file names are not breaking the file path checks, please do add the test cases for the same.